crates/swc_html_minifier/src/lib.rs

```diff
@@ -434,13 +459,128 @@ impl Minifier {
+        let default_destroy_whole = match mode {
+            CollapseWhitespaces::All => true,
+            CollapseWhitespaces::Smart | CollapseWhitespaces::Conservative => false,
+        };
+        let default_trim = match mode {
+            CollapseWhitespaces::All => true,
+            CollapseWhitespaces::Smart | CollapseWhitespaces::Conservative => false,
+        };
+
+        match namespace {
+            Namespace::HTML => {
+                match tag_name {
+                    // Inline text semantics + legacy tags + `del` + `ins` - `br`
+                    "a" | "abbr" | "acronym" | "b" | "bdi" | "bdo" | "cite" | "data" | "big"
+                    | "del" | "dfn" | "em" | "i" | "ins" | "kbd" | "mark" | "q" | "nobr" | "rp"
+                    | "rt" | "rtc" | "ruby" | "s" | "samp" | "small" | "span" | "strike"
+                    | "strong" | "sub" | "sup" | "time" | "tt" | "u" | "var" | "wbr" => {
+                        WhitespaceMinificationMode {
+                            collapse: true,
+                            destroy_whole: default_destroy_whole,
+                            trim: default_trim,
+                        }
+                    }
+                    "script" | "style" => WhitespaceMinificationMode {
+                        collapse: false,
+                        destroy_whole: true,
+                        trim: true,
+                    },
+                    "textarea" | "code" | "pre" | "listing" | "plaintext" | "xmp" => {
+                        WhitespaceMinificationMode {
+                            collapse: false,
+                            destroy_whole: false,
+                            trim: false,
+                        }
+                    }
+                    _ => WhitespaceMinificationMode {
+                        collapse: true,
+                        destroy_whole: default_destroy_whole,
+                        trim: default_trim,
+                    },
+                }
+            }
+            Namespace::SVG => match tag_name {
+                "desc" | "text" | "title" => WhitespaceMinificationMode {
+                    collapse: true,
+                    destroy_whole: true,
+                    trim: true,
+                },
+                "a" | "altGlyph" | "tspan" | "textPath" | "tref" => WhitespaceMinificationMode {
+                    collapse: true,
+                    destroy_whole: default_destroy_whole,
+                    trim: default_trim,
+                },
+                _ => WhitespaceMinificationMode {
+                    collapse: true,
+                    destroy_whole: true,
+                    trim: true,
+                },
+            },
+            _ => WhitespaceMinificationMode {
+                collapse: false,
+                destroy_whole: false,
+                trim: false,
+            },
+        }
+    }
+
+    fn is_all_whitespace(&self, data: &str) -> bool {
+        for c in data.chars() {
```

Comment:
I think this can be `!data.chars().any(is_whitespace)` instead

Reply:
@kdy1 Done :+1: